### PR TITLE
Update Build and Release workflows

### PR DESCRIPTION
**Pull request by vgrassia**
K8S-Proxy Docker image was actually using the NGINX image due to a typo.  This correct the issue.
Also, updated both Build and Release workflows using matrices for parallelization.  Removed build.sh script dependency.

### CODE REVIEW

**Review by joseph-flinn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -106,70 +124,29 @@ jobs:
...
-        shell: pwsh
+      - name: Push Docker images
+        run: |
+          docker push bitwarden/${{ matrix.service_name }}:latest
+          docker push bitwarden/${{ matrix.service_name }}:$RELEASE_VERSION
         env:
           DOCKER_CONTENT_TRUST: 1
           DOCKER_CONTENT_TRUST_REPOSITORY_PASSPHRASE: ${{ steps.retrieve-secrets.outputs.dct-delegate-2-repo-passphrase }}
 
-      - name: Make docker stub
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:172]: This should only run once (not in the matrix)

File: .github/workflows/release.yml [hunk 2]
```diff
@@ -180,13 +157,13 @@ jobs:
           touch $STUB_OUTPUT/env/uid.env
           cd docker-stub; zip -r ../docker-stub.zip *; cd ..
 
-      - name: Upload docker stub artifact
+      - name: Upload Docker stub artifact
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:160]: This should only run once (not in the matrix)

File: .github/workflows/release.yml [hunk 3]
```diff
@@ -180,13 +157,13 @@ jobs:
...
         uses: actions/upload-artifact@ee69f02b3dfdecd58bb31b4d133da38ba6fe3700
         with:
           name: docker-stub.zip
           path: ./docker-stub.zip
 
-      - name: Build swagger
+      - name: Build Swagger
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:166]: This should only run once (no in the matrix)

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -199,18 +176,17 @@ jobs:
           dotnet swagger tofile --output ../../swagger.json --host https://api.bitwarden.com `
             ./obj/Docker/publish/Api/Api.dll public
           cd ../..
-        shell: pwsh
         env:
           ASPNETCORE_ENVIRONMENT: Production
           swaggerGen: 'True'
 
-      - name: Upload swagger artifact
+      - name: Upload Swagger artifact
```
Comment (joseph-flinn) [inline .github/workflows/release.yml:183]: This should only run once (no in the matrix)

**Review by joseph-flinn — CHANGES_REQUESTED** [review]

File: .github/workflows/release.yml [hunk 5]
```diff
@@ -53,8 +54,28 @@ jobs:
           prerelease: false
 
   release:
+    name: Release
     runs-on: ubuntu-latest
     needs: setup
+    strategy:
+      fail-fast: false
+      matrix:
+        service_name: [
```
[the remark this answers is not in the document]
Comment (vgrassia) [reply .github/workflows/release.yml:63]: Array's are supported directly in the YAML spec; they are just called `sequences`. It's supported in block format (with '-' characters) or between square brackets.  I could switch it to something like below, let me know what you think:

```
- admin
- api
- attachments
- ...
```

**Review by joseph-flinn — APPROVED** [review]

**Review by joseph-flinn — APPROVED** [review]

**Pull request merged**